plugin/access.lua

```diff
@@ -108,23 +107,26 @@ function _M.run(config)
     end
 
     -- Next verify that the main cookie was received and get the access token
-    local at_cookie, err = cookie:get(config.cookie_name_prefix .. "-at")
-    if err or not at_cookie then
+    local at_cookie_name = "cookie_" .. config.cookie_name_prefix .. "-at"
+    local at_cookie = ngx.var[at_cookie_name]
+    if not at_cookie then
         ngx.log(ngx.WARN, get_error_message("No access token cookie was sent with the request", err))
         unauthorized_request_error_response(config)
     end
 
+    -- Decrypt the cookie, which is encrypted using AES256-CBC
     local access_token, err = decrypt_cookie(at_cookie, config.encryption_key)
     if err or not access_token then
         ngx.log(ngx.WARN, get_error_message("Error when decrypting access token cookie - ", err))
         unauthorized_request_error_response(config)
     end
 
-    -- For data changing requests we also expect a CSRF header to be sent with the double submit cookie
+    -- For data changing requests do double submit cookie verification in line with OWASP CSRF best practices
```

Comment:
I think it's better to do this check before decrypting cookie. Can save us some CPU cycles :)